### PR TITLE
supermatter air alarm mode

### PR DESCRIPTION
**Pull request by deathrobotpunch**
## About The Pull Request
adds a supermatter air alarm mode to all air alarms that sets internal vents to 0 pressure which rapidly replaces the air inside the SM and sets the scrubbers to expanded siphoning
## Why It's Good For The Game
this is good for the game as it adds a small bit of relief from the SM startup procedure and helps newer players startup the SM without needing out of game knowledge of how air alarm or vents work
## Changelog
:cl:
qol: added a preset to air alarms designed for supermatter chamber conditions
/:cl:

### CODE REVIEW

**Comment by deathrobotpunch** [comment]
all criticism is highly encouraged, especially after my first PR

**Comment by deathrobotpunch** [comment]
how do I fix runtimes?

**Comment by deathrobotpunch** [comment]
this will be a draft until I figure out how to fix runtimes

**Comment by lessthnthree** [comment]
At what point do we simply map in the SM complete and functional, feels like we're getting pretty close.

There should still be some pieces to learn about the engine. Couldn't there be like an instructional poster or something?

**Comment by deathrobotpunch** [comment]
> At what point do we simply map in the SM complete and functional, feels like we're getting pretty close.
> 
> There should still be some pieces to learn about the engine. Couldn't there be like an instructional poster or something?

yeah I agree, just throwing shit at a wall here but what about a holodisk instruction guide that clearly outlines the workings of the SM and tells you how to start it? they could be included in every engineers locker, though there might have to be two to get around projection range

**Comment by GuillaumePrata** [comment]
> At what point do we simply map in the SM complete and functional, feels like we're getting pretty close.

Would be the dream.

No more dumb barrier of entry for engineering that is easily bypassed by anyone watching a youtube video.
no more dumb barrier of entry that makes a slice of the players not try to play a department they might love and is essential for a round's health.
No more dumb barrier of entry that often results in a shuttle call when no one attends to it in the first 10 minutes of the round.
No more dumb chore for experienced players that need to spend 5 minutes every single shift making the same simple SM design because anything above that is a waste of time.

At this moment, I don't see any benefit to *not* have the SM online at roundstart, outside of people that enjoy laughing at new engineers fucking up the SM and ending the round 15 minutes in.

**Comment by trerri** [comment]
The "We can't the SM be optimal at roundstart!!" argument annoys me as the supermatter engine as it stands requires a comical amount of effort from a newbie engineer to set up in a way that doesn't force a shuttle call immediately. For instance, turning on the scrubbers before injecting coolant has a non-zero chance of activating the crystal on its own  (+ instant plasmafire) due to what I assume to be linda fuckery. Or what about the waste filters which have remained unchanged, completely pointless and very detrimental ever since filters were changed to have no gas type limit? The SM needs all the help that it can get, because AORN it's in a very sorry state (think Delta where you ALSO have to set up reflectors, think the stupid coolant bypass pump, think pumps in general, why do they exist??) which literally prevents players unfamiliar with engineering from playing the job because of the punishment for not knowing a huge set of (apparently) completely arbitrary setup instructions is the whole station running out of power and you getting lynched.
As far as I'm concerned, even setting the engine up every round is silly and a very ugly bandaid for a critical lack of engineering content, though I understand it's soul etc. and I don't expect it to ever go away. But on the other hand, I would really like to eventually see the engine streamlined into something comprehensible to a new player.
I'm neutral on the supermatter air alarm mode in itself.

**Comment by deathrobotpunch** [comment]
NO MORE RUNTIMES WE ARE SO BACK!

File: code/modules/atmospherics/machinery/air_alarm/air_alarm_modes.dm [hunk 1]
```diff
@@ -159,6 +158,25 @@ GLOBAL_LIST_INIT(air_alarm_modes, init_air_alarm_modes())
...
+		vent.pressure_checks = ATMOS_INTERNAL_BOUND
+		vent.internal_pressure_bound = 0
+		vent.pump_direction = ATMOS_DIRECTION_RELEASING
+		vent.update_appearance(UPDATE_ICON)
+
+	for (var/obj/machinery/atmospherics/components/unary/vent_scrubber/scrubber as anything in applied.air_scrubbers)
+		scrubber.on = TRUE
+		scrubber.filter_types = list(/datum/gas/)
+		scrubber.set_widenet(TRUE)
+		scrubber.set_scrubbing(ATMOS_DIRECTION_SCRUBBING)
```
Comment (ZephyrTFA) [inline code/modules/atmospherics/machinery/air_alarm/air_alarm_modes.dm:178]: Wrong scrubbing define.
You dont need to set filter types

Comment (deathrobotpunch) [reply code/modules/atmospherics/machinery/air_alarm/air_alarm_modes.dm]: removed the filter types

**Comment by Pickle-Coding** [comment]
Setting internal to 0 is a pointless myth that should be discouraged. It does almost nothing. All you need to do is disable the external limit, and that's it.

**Comment by lessthnthree** [comment]
> 
> At this moment, I don't see any benefit to _not_ have the SM online at roundstart, outside of people that enjoy laughing at new engineers fucking up the SM and ending the round 15 minutes in.

Downstream I made a panic button that gives engis a second chance within a specified timeframe. (Here could be 20 minutes, for example.) allowing them to try starting the engine again after some minor damage and cleanup.

I'd rather give tools that allow inexperienced engineers to learn on the job (and make mistakes) without ruining the round rather than automating away the knowledge.

**Comment by GuillaumePrata** [comment]
> I'd rather give tools that allow inexperienced engineers to learn on the job (and make mistakes) without ruining the round rather than automating away the knowledge.

This is 100% a better way to deal with things than full automation overall.
But doesn't change the fact that the SM is barrier of entry for a job vital for the round, a barrier we know a good slice of the playerbase doesn't want to learn.
It makes no sense to lock a repair job behind atmos knowledge, some people just enjoy playing engineer like a janitor, going around fixing damage so other players can have fun.

Imagine if we locked the Janitor job behind learning the SM, it is silly.

>  In the former case you have a future atmosian. In the latter you have an engineer with hours in the game but still no concept of -why- or -how- the engine works.

Now this I fully disagree, we know a lot of engineers know the bare minimum of the SM and will never bother to learn more about atmos deep knowledge.
Learning the basic SM will not push people to learn more about atmos because it is a chore, being forced to learn is just a bad experience for most people.
If we want people to learn more about atmos, it should be in their terms, they should seek knowledge because learning is FUN, not because they are forced to learn the basics of atmos just to play engineer as a janitor.

**Comment by CRITAWAKETS** [comment]
> At this moment, I don't see any benefit to _not_ have the SM online at roundstart, outside of people that enjoy laughing at new engineers fucking up the SM and ending the round 15 minutes in.

Complex SM designs require it to be inactive at roundstart in order to perform modifications to the internal SM chamber, otherwise you'll both have the pipe components slamming into the SM aswell as the airflow getting consistently disrupted. This is why it'd probably be better for the SM to be _mostly_ set up, but not fully.

**Comment by GuillaumePrata** [comment]
> Complex SM designs require it to be inactive at roundstart in order to perform modifications to the internal SM chamber...

The answer to that is simple, a way to turn the SM offline.
I made a PR for that already and while the way I went about it was not accepted, another options were suggested by the maintainer that asked to close it.

**Comment by norsvenska** [comment]
What if there was a supermatter holodeck setting? A simulated SM with no risk of screwing over the station, giving new engineers (or assistants) the ability to try their hand at setting it up.

Alternatively, after bitrunning is merged, there could be training simulations for new engineers (or potentially other staff) to teach them the ropes. 

I'm all for reducing the amount of wiki reading, but having something in place that effectively does your job for you is not the right way to go about this.

**Comment by lessthnthree** [comment]
> What if there was a supermatter holodeck setting? A simulated SM with no risk of screwing over the station, giving new engineers (or assistants) the ability to try their hand at setting it up.
> 
> Alternatively, after bitrunning is merged, there could be training simulations for new engineers (or potentially other staff) to teach them the ropes.
> 
> I'm all for reducing the amount of wiki reading, but having something in place that effectively does your job for you is not the right way to go about this.

The holodeck idea is pretty damn cool. Isn't there already an atmos holodeck program anyways?

**Comment by deathrobotpunch** [comment]
> What if there was a supermatter holodeck setting? A simulated SM with no risk of screwing over the station, giving new engineers (or assistants) the ability to try their hand at setting it up.
> 
> Alternatively, after bitrunning is merged, there could be training simulations for new engineers (or potentially other staff) to teach them the ropes.
> 
> I'm all for reducing the amount of wiki reading, but having something in place that effectively does your job for you is not the right way to go about this.

that’s actually a very good idea, though i’m not very good at mapping so someone else may need to do that, though I may make A PR draft for it and keep this up for a short while

**Comment by deathrobotpunch** [comment]
> > What if there was a supermatter holodeck setting? A simulated SM with no risk of screwing over the station, giving new engineers (or assistants) the ability to try their hand at setting it up.
> > Alternatively, after bitrunning is merged, there could be training simulations for new engineers (or potentially other staff) to teach them the ropes.
> > I'm all for reducing the amount of wiki reading, but having something in place that effectively does your job for you is not the right way to go about this.
> 
> The holodeck idea is pretty damn cool. Isn't there already an atmos holodeck program anyways?

i’m pretty sure the atmos program is emag only and is a room full of super hot plasma

**Pull request closed without merging**